### DISCUSSION TITLE
ELK +esm issue

### DISCUSSION THREAD

**Discussion by AKStratus** [mermaid-js/mermaid#4449] [Q&A] [answered]
I tried to use elk editor in my project. Unfortunately I keep getting 500 error in the console for this url: https://cdn.jsdelivr.net/npm/elkjs@0.8.2/lib/elk.bundled.js/+esm
For example https://cdn.jsdelivr.net/npm/mermaid@10.2.2/+esm and other files are accessible. This breaks elk functionality for me. 

I see that Live Editor works fine. Any Idea what needs to be done here?

**Comment by sidharthv96** [comment, marked as answer]
How are you using the script? 

From the [docs](https://mermaid.js.org/intro/n00b-gettingStarted.html#requirements-for-the-mermaid-api)...
```js
<body>
  <script type="module">
    import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs';
    mermaid.initialize({ startOnLoad: true });
  </script>
</body>
```

**Comment by AKStratus** [comment]
In .js file:
`import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10.2.2/+esm';
...
mermaid.initialize({ startOnLoad: false });`

**Comment by AKStratus** [comment]
OK, I replaced "import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10.2.2/+esm'"
with suggested "import mermaid from 'https://cdn.jsdelivr.net/npm/mermaid@10/dist/mermaid.esm.min.mjs'" and it rendered elk diagram. 

Any idea why everything else works with first import except elk diagrams? I still do not think Error 500 should be thrown in that case. Are there any obvious differences between these two approaches?

**Reply by sidharthv96** [reply]
The +esm syntax is a CDNJS specific notation. The error 500 was also thrown by the CDN, not mermaid. 

So maybe there's something wrong with how they are trying to bundle mermaid.
